tests/resources/functional/snowflake/core_engine/lca/lca_homonym.sql

```diff
@@ -0,0 +1,27 @@
+-- snowflake sql:
+select
+  ca_zip
+from (
+  SELECT
+    substr(ca_zip,1,5) ca_zip,
+    trim(name) as name,
+    count(*) over( partition by ca_zip)
+  FROM
+    customer_address
+  WHERE
+    ca_zip IN ('89436', '30868'));
+-- databricks sql:
+SELECT
+  ca_zip
+FROM
+SELECT
+  SUBSTR(ca_zip,1,5) AS ca_zip,
+  TRIM(name) AS name,
+  COUNT(*) OVER (
+    PARTITION BY
+      SUBSTR(ca_zip,1,5)
+  )
```

Comment:
I'm not sure whether this should have been dealiased or not: I believe that in Snowflake an unqualified reference such as this is assumed to be a column reference in preference to a column alias. Please test this to verify, and add a comment explaining the behaviour. Although I understand the intent of the query, I suspect that it's supposed to be interpreted as:
```suggestion
  customer_address.ca_zip
```

Reply:
> I'm not sure whether this should have been dealiased or not: I believe that in Snowflake an unqualified reference such as this is assumed to be a column reference in preference to a column alias. Please test this to verify, and add a comment explaining the behaviour. Although I understand the intent of the query, I suspect that it's supposed to be interpreted as:

The tests in [test_lca_utils.py](https://github.com/databrickslabs/remorph/blob/main/tests/unit/snow/test_lca_utils.py) do dealias window expressions. @sundarshankar89 can you opine ?

Reply:
To be clear the question is not about dealising in general within a window expression, it's to do with the ambiguity for this particular example: does the unqualified identifier (`ca_zip`) refer to the column alias or the original column?

Google suggests that in this case for Snowflake there's no way to refer to the column alias because there's a column with the same name.

I'm not sure of the full semantics for the tests in `test_lca_utils.py` but it looks like it doesn't verify output, merely whether an error occurs or not. In this case leaving the `ca_zip` reference as is would be fine and pass that test (because it's valid: it's a reference to the column).

Reply:
See this test in [test_lca_utils.py](https://github.com/databrickslabs/remorph/blob/main/tests/unit/snow/test_lca_utils.py)

```
def test_fix_lca_with_lca_in_window(normalize_string):
    input_sql = """
        SELECT
            t.col1,
            t.col2,
            t.col3 AS ca,
            ROW_NUMBER() OVER (PARTITION by ca ORDER BY t.col2 DESC) AS rn
        FROM table1 t
    """
    expected_sql = """
        SELECT
            t.col1,
            t.col2,
            t.col3 AS ca,
            ROW_NUMBER() OVER (PARTITION by t.col3 ORDER BY t.col2 DESC) AS rn
        FROM table1 AS t
    """
```

Reply:
Revisiting the spec is not in scope of this PR, but if you have doubts then we should definitely open a ticket.

Reply:
To be clear, it's about `test_query_with_same_alias_and_column_name()` from that test suite, which I don't believe specifies the expected output for this example: merely that it won't produce an error.

The output we have here seems to be new (and might be the author's _intent_ when the query was written) but I don't believe it's how Snowflake will interpret the query.

I've tried to explain above why transpilation shouldn't produce an error, and also why dealiasing _this specific example_ is incorrect for Snowflake. Maybe we should jump on a call to go over this.

Reply:
LCAs are not supported inside window clauses, so the expected translation here seems correct to me.

Reply:
I'll suggest we have a quick call in the morning, since I don't think we're understanding each other. My point is that for this specific example there's no LCA (to dealias).

Reply:
If we're going to leave the buggy behaviour in place, please change this functional test so that it represents an example that _is_ being transpiled correctly. Tests are supposed to indicate correct behaviour, and this example does not.

We already have a problem in some places where test fixtures are incorrect, and I am strongly against introducing another one.